### DISCUSSION TITLE
Can MinVer generate versions for Git Flow with tags on master branch only?

### DISCUSSION THREAD

**Discussion by mycroes** [adamralph/minver#747] [Q&A] [answered]
I'm using Git Flow with versions tagged on the master branch, however if I run `minver -t v` on develop it doesn't consider the merges to master. I understand that the merges to master are not ancestors of any commit on develop (unless I create a hotfix from master, and merge that back into develop of course). Since the documentation states that Git Flow is supported I'm wondering how one would approach this.

**Comment by adamralph** [comment, marked as answer]
The documentation states that the branching strategy is irrelevant, which is true. So yes, GitFlow is "supported".

However, MinVer relies on tags, either on the current commit or an ancestor of the current commit. That's just how it works.

**Reply by mycroes** [reply]
Thanks @adamralph, I couldn't find that exact detail in the documentation (not saying it's not in there, might have missed it just as well), this clears it up.

**Reply by adamralph** [reply]
In fact it's more accurate to say that MinVer relies on a tag on the _current_ commit. I.e. you tag the current commit in order to build and release. When the current commit is _not_ tagged, the ancestors are examined to generate a "filler" version using the height. These versions are not intended for release. This is all explained in more detail in _[How it works](https://github.com/adamralph/minver#how-it-works)_.
